repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2748,28 +2751,66 @@ public List<String> propagateClassification(String entityGuid, String classifica
                 return null;
             }
 
-            List<String> impactedVerticesGuidsToLock = impactedVertices.stream().map(x -> GraphHelper.getGuid(x)).collect(Collectors.toList());
-            GraphTransactionInterceptor.lockObjectAndReleasePostCommit(impactedVerticesGuidsToLock);
+            List<String> propagatedEntitiesGuid = new ArrayList<>();
+            long impactedVerticesSize = impactedVertices.size();
+            int offset = 0;
 
-            AtlasClassification classification       = entityRetriever.toAtlasClassification(classificationVertex);
-            List<AtlasVertex>   entitiesPropagatedTo = deleteDelegate.getHandler().addTagPropagation(classificationVertex, impactedVertices);
+            LOG.info(String.format("Total number of vertices to propagate: %d", impactedVerticesSize));
+            AtlasPerfMetrics.MetricRecorder chunkedPropMetricRecorder = RequestContext.get().startMetricRecord("chunkedPropagationAndNotification");
 
-            if (CollectionUtils.isEmpty(entitiesPropagatedTo)) {
-                return null;
-            }
+            do {
+                int toIndex;
+                if (offset + CHUNK_SIZE > impactedVerticesSize) {
+                    toIndex = (int) impactedVerticesSize;
+                }
+                else {
+                    toIndex = offset + CHUNK_SIZE;
+                }
+                List<String> chunkedGuids = processChunkedPropagation(impactedVertices.subList(offset, toIndex), classificationVertex);
 
-            List<AtlasEntity> propagatedEntities = updateClassificationText(classification, entitiesPropagatedTo);
+                if((chunkedGuids != null) && (! chunkedGuids.isEmpty())){
```

Comment:
is it possible that this is null here because addTagPropagation or some other logic returned null without throwing exception? in that case this logic will continue chunking even though one of the chunks returned null

Reply:
Due to the line 2797, It can return null.